.github/workflows/backend.yml

```diff
@@ -0,0 +1,110 @@
+name: backend-py3
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  backend-py3:
+    runs-on: ubuntu-16.04
+    continue-on-error: true
+    strategy:
+      matrix:
+        instance: [0, 1, 2, 3]
+
+    env:
+      SENTRY_PYTHON3: 1
+      PIP_DISABLE_PIP_VERSION_CHECK: on
+
+      SENTRY_LIGHT_BUILD: 1
+      SENTRY_SKIP_BACKEND_VALIDATION: 1
+
+      # Use this to override the django version in the requirements file.
+      DJANGO_VERSION: ">=1.11,<1.12"
+
+      PYTEST_SENTRY_DSN: https://6fd5cfea2d4d46b182ad214ac7810508@sentry.io/2423079
+      # XXX(py3): Reruns are disabled for until all tests are passing.
+      PYTEST_ADDOPTS: ""
+      PYTEST_SENTRY_ALWAYS_REPORT: no
+
+      # services configuration
+      SENTRY_REDIS_HOST: redis
+      DATABASE_URL: postgresql://postgres:postgres@localhost/sentry
+
+      # Number of matrix instances
+      TOTAL_TEST_GROUPS: ${{ strategy.job-total }}
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up outputs
+        id: config
+        env:
+          MATRIX_INSTANCE: ${{ matrix.instance }}
+        run: |
+          # XXX(py3): Minors that aren't latest seem to not be available.
+          echo "::set-output name=python-version::3.6.11"
+          echo "::set-output name=matrix-instance-number::$(($MATRIX_INSTANCE+1))"
+
+      - name: Set up Python ${{ steps.config.outputs.python-version }}
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ steps.config.outputs.python-version}}
+
+      - name: Install pip
+        run: |
+          pip install --no-cache-dir --upgrade "pip>=20.0.2"
+
+      - name: Get pip cache dir
+        id: pip-cache
+        run: |
+          echo "::set-output name=dir::$(pip cache dir)"
+
+      - name: pip cache
+        uses: actions/cache@v1
+        with:
+          path: ${{ steps.pip-cache.outputs.dir }}
+          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements-*.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-
+
+      - name: Install System Dependencies
```

Comment:
Shouldn't this come _before_ pip cache?

Reply:
I'll defer to @billyvg here, I think ideally it'd be the 1st step, even before actions/checkout, but not sure what convention is.

Reply:
Yeah I think that makes sense

Reply:
Ok, I can submit a PR for this later.